pkg/chunk/iterator.go

```diff
@@ -33,8 +35,11 @@ const BatchSize = 12
 // Batch is a sorted set of (timestamp, value) pairs.  They are intended to be
 // small, and passed by value.
 type Batch struct {
-	Timestamps [BatchSize]int64
-	Values     [BatchSize]float64
-	Index      int
-	Length     int
+	Timestamps      [BatchSize]int64
+	Values          [BatchSize]float64
+	Histograms      [BatchSize]*histogram.Histogram
+	FloatHistograms [BatchSize]*histogram.FloatHistogram
```

Comment:
Any idea how I can change to make Batch smaller? It should be only 1 value type per batch and other array are redundant.